docs/source/install-script.rst

```diff
@@ -0,0 +1,141 @@
+
+Linux
+~~~~~
+
+For Linux distributions the system dependencies are installed via the :bash:`apt-get` package manager. The necessary dependencies for running default firedrake are:
+
+- build-essential
+- autoconf
+- automake
+- bison
+- flex
+- cmake
+- gfortran
+- git
+- libtool
+- python3-dev
+- python3-pip
+- python3-tk
+- python3-venv
+- zlib1g-dev
+- libboost-dev
+- libopenblas-dev (if no configuration is provided by the user)
+
+As a small observation, packages like *bison* and *flex* are used for *PT-Scotch* (tool for efficient parallel graph ordering).
+
+MacOS
+~~~~~
+
+For MacOs distributions (based on Darwin) the install script uses the :bash:`homebrew` package manager to install the following necessary dependencies:
+
+- `gcc`
+- `autoconf`
+- `make`
+- `automake`
+- `cmake`
+- `libtool`
+- `boost`
+- `openblas` (if no configuration is provided by the user)
+
+Additionally the install script will also want to install xcode-command line tools.
+
+After the installation of the system requirements is done, the installation script will then create a virtual python environment and prepare itself to install all its software dependencies there.
+
+Building and configuring PETSc
+-------------------------------
+
+One important package on which Firedrake software is built is the `PETSc_ <https://petsc.org/release/>`_ library, which offers a toolkit when dealing with partial differential equations. 
+The install script enables some customization for installing the PETSc, but also offers the user the option to use a previous installation of the library via the parameter :bash:`hounour_petsc`.
+If no previous previous installation is provided, the installation script will then build the PETSc package from source in the *src*  directory within the virtual python environment.
+
+The basic options for installing the library using the install argument :bash:`--minimal_petsc` are:
+
+.. code-block:: python
+
+   petsc_options = { "--with-fortran-bindings=0",
+                     "--with-debugging=0",
+                     "--with-shared-libraries=1",
+                     "--with-c2html=0",
+                     "--download-eigen=%s/src/eigen-3.3.3.tgz " % firedrake_env,
+                     # File format
+                     "--download-hdf5",
+                     # AMG
+                     "--download-hypre",
+                     # Sparse direct solver
+                     "--download-superlu_dist",
+                     # Parallel mesh partitione[]()r
+                     "--download-ptscotch",
+                     # For superlu_dist amongst others.
+                     "--with-cxx-dialect=C++11"}
+
+This option is useful if we want to have a faster build and therefore is very useful for testing.
+For a more detail explanation of the parameters please check the PETSc install documentation (page)[https://petsc.org/release/install/].
+
+If no minimal version of PETSc is required (:bash:`--minimal_petsc false`):
```

Comment:
I'm not sure what you mean by `--minimal_petsc false`. To have a non-minimal install you just need to omit the flag.

Reply:
Yeah you're right. I will fix this. (I missinterpreted the parse arguments of the install script)